### ISSUE TITLE
Update main navigation focus styles

### ISSUE THREAD

**Issue by alizedebray** [swisspost/design-system#6924]
The focus outline of the main navigation items must be improved in order to be visible for any state of the navigation items. 

Currently the outline has no offset, but it's not visible with selected items: 

<img width="1285" height="70" alt="Image" src="https://github.com/user-attachments/assets/0f8cebf1-bb0c-4d33-95e7-fdba0b03f816" />

<img width="1285" height="70" alt="Image" src="https://github.com/user-attachments/assets/8cb1ba7e-1562-4d18-8f39-6192b83924fb" />

On the other hand, if we add an offset as designed in Figma, the focus ring is barely visible:

<img width="1285" height="70" alt="Image" src="https://github.com/user-attachments/assets/0debca69-87bb-401a-a3fb-e2d72773c973" />

The second option better match with [WCAG SC 2.4.11](https://www.w3.org/WAI/WCAG22/Understanding/focus-not-obscured-minimum):
> In recognition of the complex responsive designs common today, this AA criterion allows for the component receiving focus to be partially obscured by other author-created content. A partly obscured component can still be very visible, although the more of it that is obscured, the less easy it is to see. For that reason, authors should attempt to design interactions to reduce the degree and frequency with which the item receiving focus is partly obscured. For best visibility, none of the component receiving focus should be obscured. This preferred outcome is covered by the AAA criterion [Focus Not Obscured (Enhanced)](https://www.w3.org/WAI/WCAG22/Understanding/focus-not-obscured-enhanced.html).

But we could improve the focus even more to make it fully visible all the time.

**Comment by gfellerph** [comment]
Pattern discussion: revisit focus styles in the header.

**Comment by myrtasakellariou** [comment]
There is also this case of the Microsite mobile header where the focus ring gets cut off. 

<img width="582" height="376" alt="Image" src="https://github.com/user-attachments/assets/1b007652-6dc2-46e6-9879-18fc9edad450" />

**Comment by sandra-post** [comment]
@alizedebray  Redesigned focus ring for header [nav links](https://www.figma.com/design/JIT5AdGYqv6bDRpfBPV8XR/Foundations---Components-Next-Level?node-id=2618-27381&t=Jbos8rzl4f5HAq5d-4) and [megamenu items](https://www.figma.com/design/JIT5AdGYqv6bDRpfBPV8XR/Foundations---Components-Next-Level?node-id=7402-44004&t=Jbos8rzl4f5HAq5d-4), changes:

**Focus ring – GENERAL**

- Offset
  - OLD: X = 2 | Y = 2
  - NEW: X = -2 | Y = -2
    - New token for DEV: `post.focus.outline.offset2` (`post.device.spacing.padding.28` --> `post.core.dimension.n2`)
- Border alignment
  - OLD: Outside
  - NEW: Inside

**Focus ring – SELECTED**

- Color
  - OLD: `post.focus.outline.color`
  - NEW: `post.focus.outline.color-inverted`
